cohortextractor/date_utils.py

```diff
@@ -0,0 +1,71 @@
+import calendar
+import datetime
+
+
+def generate_date_range(date_range_str):
```

Comment:
This is ported from V1 almost entirely, with the addition of a check for valid end of month dates on line 61